howto/android/odm_vhal.md

```diff
@@ -0,0 +1,128 @@
+*since 1.22.0*
+
+[note type="information" status="Note"]Replacing the Anbox VHAL is only supported on AAOS images.
+The Anbox Cloud dashboard does not support custom VHAL implementations.[/note]
+
+This document will guide through the process of replacing the Anbox Cloud VHAL
+implementation with your own implementation placed in the
+[ODM partition](https://source.android.com/docs/core/architecture/partitions/odm-partitions)
+using an [addon](https://anbox-cloud.io/docs/explanation/addons).
+
+Your custom VHAL implementation must provide an
+[HIDL VHAL interface](https://source.android.com/docs/automotive/vhal/hidl-vhal-interface)
+and be built against VNDK 29.
+
+## Create the addon
+
+This addon will be tasked with the following:
+
+1. Copying the VHAL binary to `/odm/bin/hw`
+1. Copying an
+[init file](https://android.googlesource.com/platform/system/core/+/master/init/README.md)
+for your VHAL service to `/odm/etc/init`
+1. Disabling the Anbox Cloud VHAL by setting the `ro.anbox.automotive.vhal`
+property to `odm`. This will be done by writing that system property to the
+`/odm/etc/build.prop` file.
+
+In the following, we will name our addon `custom-vhal`.
+To override the VHAL implementation, we will use a
+[pre-start hook](https://anbox-cloud.io/docs/reference/hooks) which is executed
+before Android gets started.
+
+The directory layout for the addon is strict and must be:
+
+```
+custom-vhal/
+├── hooks
+│   └── pre-start
+├── manifest.yaml
+├── vhal
+└── vhal.rc
+```
+
+`vhal` is the VHAL implementation built in the previous section. If your VHAL
```

Comment:
"built in the previous section" is a bit confusing for me as we have not provided instructions to build it. IMO, it is a prerequisite to have a cusom implementation ready to do this procedure. Let me know if my understanding is wrong.

```suggestion
`vhal` is your custom VHAL implementation.
```

Reply:
Oh you're right, that's a leftover from a previous version where I assumed I would have instructions for building the binary... Thanks for catching this!